spec/integration/lookup_spec.rb

```diff
@@ -134,4 +134,32 @@
       )
     end
   end
+
+  context 'with plan_hiera' do
+    let(:hiera_config) { File.join(boltdir, 'plan_hiera.yaml') }
+    let(:plan)         { 'test::plan_lookup' }
+    let(:uri)          { 'localhost' }
+
+    it 'uses plan_hierarchy outside apply block, and hierarchy in apply block' do
+      pending('Until Puppet 6.18 is released')
+      result = run_cli_json(cli_command + %W[-t #{uri}])
+      expect(result['outside_apply']).to eq('goes the weasel')
+      expect(result['in_apply'].keys).to include('Notify[tarts]')
+    end
+  end
+
+  context 'with invalid plan_hierarchy' do
+    let(:hiera_config) { File.join(boltdir, 'plan_hiera_interpolations.yaml') }
+    let(:plan)         { 'test::plan_lookup' }
+    let(:uri)          { 'localhost' }
+
+    it 'raises a validation error' do
+      pending('Until Puppet 6.18 is released')
```

Comment:
I think using pending without having the rest of the test as a "block" argument to it will result in the "end" of that step (so the expected error code wouldn't run). I thought the use of pending was to put the code you *expect* to work in the block, each time the test is run, if that code fails the test can still "pass" as pending, however as soon as the expectations in the block are "true" the test fails and tells you that your thing is no longer pending.

Reply:
No, this will correctly pend them. This is the output if I use my Puppet PR:
```
lucy ~/githubs/bolt (GH-1835) $ be rspec spec/integration/lookup_spec.rb 
Run options: exclude {:sudo=>true, :windows=>true, :windows_agents=>true}
..............FF

Failures:

  1) lookup() in plans with plan_hiera uses plan_hierarchy outside apply block, and hierarchy in apply block FIXED
     Expected pending 'Until Puppet 6.18 is released' to fail. No error was raised.
     # ./spec/integration/lookup_spec.rb:143

  2) lookup() in plans with invalid plan_hierarchy raises a validation error FIXED
     Expected pending 'Until Puppet 6.18 is released' to fail. No error was raised.
     # ./spec/integration/lookup_spec.rb:156

Finished in 5.1 seconds (files took 1.56 seconds to load)
16 examples, 2 failures

Failed examples:

rspec ./spec/integration/lookup_spec.rb:143 # lookup() in plans with plan_hiera uses plan_hierarchy outside apply block, and hierarchy in apply block
rspec ./spec/integration/lookup_spec.rb:156 # lookup() in plans with invalid plan_hierarchy raises a validation error
```

Reply:
ok, so when puppet is release you will need to update the tests. I was thinking of a use case where you write the tests that *should* pass (when the thing that is pending is released) and when they start passing (pending starts "failing") you simply remove the pending. I did not realize the code after a `pending()` executed.

Reply:
Yup, these will start failing when Puppet is released and we'll need to remove the `pending`.

Reply:
Ahhh, I see. I totally missed that the *desired* outcome is the Error. Ignore me!